lib/engine/game/g_1854/step/track_and_token.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/track_and_token'
+
+module Engine
+  module Game
+    module G1854
+      module Step
+        class TrackAndToken < Engine::Step::TrackAndToken
```

Comment:
The `TrackAndToken` is for when a corporation can lay track and place station tokens in any order. In 1854 you have to place track before you place any station tokens, so I think that you will need separate track and token steps. Or am I not be understanding what you're doing here? It's a bit difficult to fully review this without seeing the round definitions in the G1854::Game class.

Reply:
This is for the part of the game where you merge two number minors into one of the ABC corporations.  It's only used to keep track of the need to pick a hex -> possibly lay tile -> then pick a city for home token.

Reply:
OK.
I think this might be more complicated than it needs to be. This could be split into two steps:

1. A subclass of `Step::HomeToken` where the president of the Lokahlbahn chooses the city for the home token (you will need to do something with `home_token_can_be_cheater` for the case where there is no available slot on the yellow tile in Amstetten). This would remove the need for the assign action, and all the code to do with pending tokens.
2. A subclass of `Step::Track` to allow the placement of the green tile in the home hex.

Reply:
I think I'd prefer to leave that for a future pr if possible so I can keep track of it more easily. Seems like a relatively big change.

Reply:
I expect that you'd end up with less code if you did it this way, as you've got a lot of code in this step copied from `Step::HomeToken` that would no longer be needed.
But this can be left for later if you want. I don't think that there's anything wrong with this code, it just could be made simpler.